### PR TITLE
Feature/kernels

### PR DESCRIPTION
**Pull request by thiagopbueno**
Add kernel metrics in `mapo/kernels/__init__.py` to compare gradients in dynamics loss.

### CODE REVIEW

File: mapo/agents/mapo/losses.py [hunk 1]
```diff
@@ -0,0 +1,130 @@
...
+    )
+    return -tf.reduce_mean(model.compute_states_log_prob(obs, actions, next_obs))
+
+
+def dynamics_pga_loss(batch_tensors, model, actor_loss, config):
+    """Compute Policy Gradient Aware dynamics loss"""
+    gmapo = tf.gradients(actor_loss, model.actor_variables)
+    dpg_loss = actor_dpg_loss(batch_tensors, model)
+    dpg = tf.gradients(dpg_loss, model.actor_variables)
+    kernel = KERNELS[config["kernel"]]
```
Comment (thiagopbueno) [inline mapo/agents/mapo/losses.py]: As you can see I passed the kernel as a string (l2, l1, cos_similarity, ...)... I didn't see the use of passing the lambda itself with ray.tune.function... if this is not fine, just let me know.

File: mapo/agents/mapo/mapo.py [hunk 2]
```diff
@@ -10,11 +10,20 @@
...
         # How many samples to draw from the dynamics model
         "branching_factor": 1,
+        # Whether to use the env's dynamics to calculate the actor loss
+        "use_true_dynamics": False,
+        # Which model-learning optimization to use
+        # Valid values: "mle" (Maximum Likelihood Estimation),
+        # "pga" (DPG-aware loss function),
+        "model_loss": "mle",
+        # Which kernel metric to compare gradients in dynamics loss
+        "kernel": "l2",
```
Comment (thiagopbueno) [inline mapo/agents/mapo/mapo.py]: As a default kernel I'm currently using "l2", but we can change this anytime...

**Comment by thiagopbueno** [comment]
BTW... this branch cannot be automatically merged because I already merged it with develop (after I accepted your PR of the feature/pga-model-learning) to start running experiments...

File: mapo/agents/mapo/losses.py (hunk 1 above)
Comment (0xangelo) [reply mapo/agents/mapo/losses.py]: Nice! I agree it's better now.

File: mapo/agents/mapo/mapo.py (hunk 2 above)
Comment (0xangelo) [reply mapo/agents/mapo/mapo.py]: No problem!

File: mapo/agents/mapo/off_mapo.py [hunk 3]
```diff
@@ -14,6 +14,17 @@
...
+        # How many samples to draw from the dynamics model
+        "branching_factor": 1,
+        # Whether to use the env's dynamics to calculate the actor loss
+        "use_true_dynamics": False,
+        # Which model-learning optimization to use
+        # Valid values: "mle" (Maximum Likelihood Estimation),
+        # "pga" (DPG-aware loss function),
+        "model_loss": "mle",
+        # Which kernel metric to compare gradients in dynamics loss
+        "kernel": "l2",
```
Comment (0xangelo) [inline mapo/agents/mapo/off_mapo.py]: We don't need to repeat configs declared under `mapo.agents.mapo.mapo.DEFAULT_CONFIG`, as `merge_dict` above makes sure both on- and -off-policy versions share configs. 

IMO, the only configs that should be declared here are ones not present in the on-policy version (such as exploration strategies) or overwrites for hyperparameters which we find reasonable after testing.

**Review by 0xangelo — CHANGES_REQUESTED** [review]
LGTM! The only thing remaining is to rebase this branch w.r.t. `develop` and force push once the conflicts are resolved.

File: mapo/agents/mapo/off_mapo.py (hunk 3 above)
Comment (thiagopbueno) [reply mapo/agents/mapo/off_mapo.py]: ok. noted! gonna change that.
